### DISCUSSION TITLE
Lightbox Gallery( 点击查看大图 )

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1598] [插件说明]
描述
---

> 在阅读模式下，点击任意图片，进入 **图库模式**。

地址
---

[https://simpread.ksria.cn/plugins/details/TmZSYtz5lA](https://simpread.ksria.cn/plugins/details/TmZSYtz5lA)

截图
---

![image](https://user-images.githubusercontent.com/81074/124442620-82e51a80-ddaf-11eb-8c5a-809709d8cf25.png)

视频
---

![2021-07-05_16-34-38 (1)](https://user-images.githubusercontent.com/81074/124444815-91343600-ddb1-11eb-966e-e812c57e58aa.gif)

全新 2.0 版
---

全新 2.0 版彻底解决了标注无法使用的问题 **[1.0 版 已被废弃](https://simpread.ksria.cn/plugins/details/VQOZdNET2d)**，已安装 1.0 版用户请 **[重新安装 2.0 版](https://simpread.ksria.cn/plugins/details/TmZSYtz5lA)**，这两个插件的数据是一致的，无需重新设置。

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1088" alt="image" src="https://github.com/user-attachments/assets/25800e00-b49b-43a1-ace8-93e01504ffb4">

功能
---

- 图库模式

    - 查看当前图片的大图

    - 底部浏览全部图片

    - 顶部工具条，功能包括：下载当前图片 · 新标签页打开 · 全部下载 · 复制到剪切板 · 播放 · 关闭

- [限制图片最大高度](https://github.com/Kenshin/simpread/discussions/1598#discussioncomment-3480036)

- [优化图片外部链接打开](https://github.com/Kenshin/simpread/discussions/1598#discussioncomment-3473348)

- 进入图库模式后，支持 **鼠标滚轮放大/缩小图片**

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/TmZSYtz5lA) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 2.0.0

- `important` 全新 2.0 版彻底解决了标注无法使用的问题 **[1.0 版 已被废弃](https://simpread.ksria.cn/plugins/details/VQOZdNET2d)**，已安装 1.0 版用户请 **[重新安装 2.0 版](https://simpread.ksria.cn/plugins/details/TmZSYtz5lA)**，这两个插件的数据是一致的，无需重新设置。

- `Add` 鼠标滚轮放大/缩小图片

- `Optimize` 选项卡升级到 UI 2.0 方案

- `Fix` 图片监控行为（使用 [稍后读辅助增强插件](https://github.com/Kenshin/simpread/discussions/5943) 会导致此插件失效，新版解决了此问题）

### 1.0.4

- `Add` 手动调整图片尺寸

- `Optimize` 优化图片最大高度的 CSS

### 1.0.3

- `Add` [图片最大高度白名单](https://github.com/Kenshin/simpread/discussions/1598#discussioncomment-6370658)

### 1.0.2

- `Add` [限制图片最大高度](https://github.com/Kenshin/simpread/discussions/1598#discussioncomment-3480036)

- `Add` [优化图片外部链接打开](https://github.com/Kenshin/simpread/discussions/1598#discussioncomment-3473348)

- `Fix` 在某些页面无法使用的问题

### 1.0.1

- `Fix` 某些页面下 CSS 错误。

### 1.0.0

- `Add` 图库模式

**Comment by Dirushui** [comment]
希望可以出一个鼠滚轮放大缩小图片的功能

**Reply by Kenshin** [reply]
谢谢建议，其实这个功能的升级计划放到 Todo 里面好久好久了，无奈一直没太多时间来做这些。
预计下个版本会集中时间升级一批插件。  😂

**Reply by Kenshin** [reply]
嗨

我升级了 Lightbox → Lightbox Gallery，升级内容和讨论请前往 https://github.com/Kenshin/simpread/discussions/1598

本次升级，包含了很多功能：

- 查看当前图片的大图
- 底部浏览全部图片
- 顶部工具条
    - 下载当前图片
    - 新标签页打开
    - 全部下载
    - 复制到剪切板
    - 播放
    - 关闭

**Reply by felix2049lou** [reply]
https://leadingmrk.com/the-most-complete-guide-to-notion/#notion-quick-start
即使我把标注功能都关了，插件对此网站的图片一点效果都没有效

**Reply by Kenshin** [reply]
> https://leadingmrk.com/the-most-complete-guide-to-notion/#notion-quick-start 即使我把标注功能都关了，插件对此网站的图片一点效果都没有效

仅这一个网站吗？其它的没有这个问题，是吗？

**Comment by Kenshin** [comment]
# 优化图片外部链接打开

部分页面的图片外层会加入一个 `a` 标签，当点击图片时会新标签页打开，但这种方式 [会影响标注](https://github.com/BigSharkLx/front-end-interview/blob/main/7%20offer%E6%94%B6%E5%89%B2%E6%9C%BA%E4%B9%8B%E5%89%8D%E7%AB%AF%E5%B7%A5%E7%A8%8B%E5%8C%96%E7%AF%87.md)，所以增加了一个设置来自动删除这种情况的父类链接。

## 设置

![image](https://user-images.githubusercontent.com/81074/186826063-6361c841-f96c-449a-8d0e-631299b586ed.png)

## 说明

当前开启 **自动启用** 后，全部页面都会自动替换。或者也可以使用白名单，支持

- 全 URL e.g. `https://sspai.com/post/69972`
- 主域名 e.g. sspai.com （会自动忽略 `www` ）

一个例子

```
github.com
https://sspai.com/post/69972
```

**Comment by Kenshin** [comment]
# 限制图片最大高度

当浏览器尺寸较小，且图片的长度较大时，[会影响浏览效果](https://github.com/Kenshin/simpread/issues/4367)，因此可以使用定制最大化高度的方式来限制这种现象。

## 设置

![image](https://user-images.githubusercontent.com/81074/186825619-2665c60f-ee6c-4d58-a1b1-78d023158bbc.png)

## 手动优化

因为有些插件会在进入阅读模式后优化界面，例如 [微信公众号辅助增强](https://github.com/Kenshin/simpread/discussions/5851) 而这时此功能会失效，所以 1.0.4 版增加了手动方式，位置在：右下角 → 动作 → 插件触发器

<img width="472" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/d534f5b3-7a12-453a-801a-2a6e6f6a9b9f">

使用后 **无论当前页面是否在白名单都会执行图片最大高度限制**。

## 注意

因为使用了 `max-height` 所以 **不支持百分比**。

**Reply by Kenshin** [reply]
# 白名单

这是 1.0.3 版增加的功能，

<img width="451" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/41a70bda-6c93-404e-a9d9-7e645bca3517">

因为有些页面使用此功能后效果不好，所以增加了白名单功能，毕竟高度远大于宽度的图片相对较少，所以当遇到此页面时，可以将这个页面单独加入到白名单中。

## 支持

- 全 URL e.g. `https://www.pingwest.com/a/284783`
- 主域名 e.g. sspai.com （会自动忽略 `www` ）

一个例子

```
sspai.com
https://www.pingwest.com/a/284783
```

也就是当访问 `sspai.com` 或者 `https://www.pingwest.com/a/284783s` 后会自动优化 **图片最大高度**。
